addons/l10n_latam_check/models/account_payment.py

```diff
@@ -0,0 +1,286 @@
+                            if rec.payment_type == 'outbound' else rec.destination_journal_id.name))
+                elif rec.payment_type == 'inbound' and not rec.is_internal_transfer and \
+                        rec.l10n_latam_check_id.l10n_latam_check_current_journal_id:
+                    raise ValidationError(_("Check '%s' is on journal '%s', we can't receive it again") % (
+                        rec.l10n_latam_check_id.display_name, rec.journal_id.name))
+
+        res = super().action_post()
+
+        # mark own checks that are not printed as sent
+        for rec in self.filtered(lambda x: x.check_number):
+            sequence = rec.l10n_latam_checkbook_id.sequence_id
+            sequence.sudo().write({'number_next_actual': int(rec.check_number) + 1})
+            rec.write({'is_move_sent': True})
+        return res
+
+    @api.onchange('payment_method_line_id', 'is_internal_transfer', 'journal_id', 'destination_journal_id')
+    def reset_check_ids(self):
+        """ If any of this fields changes the domain of the selectable checks could change """
+        self.l10n_latam_check_id = False
+
+    @api.onchange('check_number')
+    def _onchange_check_number(self):
+        for rec in self.filtered(lambda x: x.journal_id.company_id.country_id.code == "AR"):
+            if rec.check_number and rec.check_number.isdecimal():
+                rec.check_number = '%08d' % int(rec.check_number)
+
+    @api.depends('l10n_latam_check_operation_ids.state')
+    def _compute_l10n_latam_check_current_journal(self):
+        new_checks = self.filtered(lambda x: x.payment_method_line_id.code == 'new_third_checks')
+        if not new_checks:
+            return
+
+        groups = self.env['account.payment'].read_group(
+            [('l10n_latam_check_id', 'in', new_checks.ids), ('state', '=', 'posted')],
+            ['is_internal_transfer:array_agg', 'payment_type:array_agg', 'journal_id:array_agg',
+                'paired_internal_transfer_payment_id:array_agg', 'date:array_agg'],
+            ['l10n_latam_check_id', 'date:day', 'id:to_sort_by_id'], orderby="date desc, id desc", lazy=False)
+
+        checks_mapping = {}
+        for group in groups:
+            l10n_latam_check_id = group['l10n_latam_check_id'][0]
+            if l10n_latam_check_id not in checks_mapping:
+                checks_mapping[l10n_latam_check_id] = group
+
+        for rec in new_checks:
+            if rec.id not in checks_mapping:
+                rec.l10n_latam_check_current_journal_id = rec.journal_id
+                continue
+            check_operations = checks_mapping.get(rec.id)
+            if check_operations.get('is_internal_transfer')[0] and check_operations.get('payment_type')[0] == 'outbound':
+                rec.l10n_latam_check_current_journal_id = rec.browse(
+                    check_operations.get('paired_internal_transfer_payment_id')[0]).journal_id
```

Comment:
get( , []) then or something?

Reply:
read_group changed for .search and this was improoved there